### ISSUE TITLE
Dashboards: Relative link to dashboard causes incorrect 404

### ISSUE THREAD

**Issue by mure** [grafana/grafana#83443]
### What happened?

https://github.com/grafana/grafana/assets/9257800/8bba6341-2d64-4d84-afdf-0b0be73d0e18

I created a panel link to another dashboard using this format: `../d/<dashboard_uid>/<dashboard_name>`. After navigating to the link, a 404 is returned, even though the URL seems correct. Refreshing the page then shows the linked dashboard.

### What did you expect to happen?

If the constructed URL of the link is correct (which it is), it should display the dashboard on the first load. A refresh shouldn't be required.

### Did this work before?

Not that I know of. 

### How do we reproduce it?

The video and description should be sufficient to reproduce.

### Is the bug inside a dashboard panel?

n/a

### Environment (with versions)?

Grafana: 10.0.3 (also tested on latest 10.x)
OS: MacOS
Browser: Chromium based

### Grafana platform?

None

**Comment by evictorero** [comment]
Thanks or reporting this issue, I could reproduce it. It only happens when opening the relative URL in the same tab.

**Comment by Letra103622** [comment]
I would like to be assigned to this issue if possible

**Comment by joshhunt** [comment]
@mure I've been looking into this, while reviewing https://github.com/grafana/grafana/pull/85646, and have some questions:

 - It seems like `../../d/uid/title` should be the correct form for a relative link like this? (note this still has the same issue you're describing, but I think it should be the most correct URL)
 - What's the case for using a relative path like this, rather than the absolute `/d/uid/title`. Absolute paths like this still work with serving grafana from a subpath.

**Comment by mure** [comment]
@joshhunt 

Agreed that `/d/uid/title` is the preferred format. That is the workaround I provided to the user who reported this. But it still seemed like a valid issue that Grafana returns a 404 even though the link is to the correct URL.
